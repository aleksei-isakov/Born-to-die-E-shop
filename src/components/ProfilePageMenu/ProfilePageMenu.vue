<template>
  <div>
    <profile-page-menu-desktop v-if="isDesktop" />
    <profile-page-menu-mobile v-else />
  </div>
</template>

<script>
import ProfilePageMenuMobile from './ProfilePageMenuMobile/ProfilePageMenuMobile.vue';
import ProfilePageMenuDesktop from './ProfilePageMenuDesktop/ProfilePageMenuDesktop.vue';
import { mapGetters } from 'vuex';

export default {
  name: 'ProfilePageMenu',

  components: {
    ProfilePageMenuMobile,
    ProfilePageMenuDesktop
  }
};
</script>
