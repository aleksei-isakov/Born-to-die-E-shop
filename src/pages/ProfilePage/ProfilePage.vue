<template>
  <my-profile-info />
</template>

<script>
import MyProfileInfo from '@/components/MyProfileInfo/MyProfileInfo.vue';

export default {
  components: {
    MyProfileInfo
  }
};
</script>
