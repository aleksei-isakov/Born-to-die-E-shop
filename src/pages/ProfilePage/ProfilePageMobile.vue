<template>
  <div class="profile-page__container">
    <div class="profile-page__wrapper">
      <div class="profile-page__header">
        <h1>Profile</h1>
      </div>
      <div class="profile-page__sections">
        <div class="profile-page__sidebar">
          <div class="profile-page__sidebar_info">
            <div class="profile-page__sidebar_item">
              <base-button-router path="/profile">
                <i class="fas fa-house-user fa-2x"></i>
                Profile
              </base-button-router>
            </div>
            <div class="profile-page__sidebar_item">
              <base-button-router path="/wishlist">
                <i class="fas fa-heart fa-2x"></i>
                Wishlist
              </base-button-router>
            </div>
            <div class="profile-page__sidebar_item">
              <base-button-router path="/orders">
                <i class="fas fa-folder fa-2x"></i>
                Orders
              </base-button-router>
            </div>
            <div class="profile-page__sidebar_item">
              <base-button-router path="/address-book">
                <i class="fas fa-address-book fa-2x"></i>
                Address book
              </base-button-router>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter';
export default {
  name: 'ProfilePageMobile',
  components: {
    BaseButtonRouter
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

@media (max-width: $tablet-size) {
  .profile-page {
    &__header {
      display: flex;
      align-self: flex-start;
      padding: 40px;

      h1 {
        font-weight: normal;
        color: $font-color-subtitle;
      }
    }

    &__sidebar {
      position: absolute;
      bottom: 0;
      display: flex;
      align-items: center;
      justify-content: center;
      width: 100%;
      background: $primary;
      min-height: 80px;
    }

    &__sidebar_info {
      width: 100%;
      display: flex;
      align-items: space-around;
    }

    &__sidebar_item {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      width: 25%;
      height: 100%;
      i {
        color: #ffffff;
      }
      .base-button-router {
        display: flex;
        color: #ffffff;
        opacity: 50%;
        &:hover {
          text-decoration: none;
          color: #ffffff;
          opacity: 100%;
        }
      }
    }
  }
}
</style>
