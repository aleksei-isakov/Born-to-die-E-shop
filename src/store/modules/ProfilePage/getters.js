const getters = {
  isDesktop: (state) => state.isDesktop
};

export default getters;
