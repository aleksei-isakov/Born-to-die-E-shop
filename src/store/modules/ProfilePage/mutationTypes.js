export default {
  SWITCH_TO_MOBILE: 'SWITCH_TO_MOBILE',
  SWITCH_TO_DESKTOP: 'SWITCH_TO_DESKTOP'
};
