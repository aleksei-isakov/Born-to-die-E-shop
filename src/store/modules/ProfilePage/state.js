const state = {
  isDesktop: true
};

export default state;
