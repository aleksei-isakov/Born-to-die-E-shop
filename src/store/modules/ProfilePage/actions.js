import mutationTypes from './mutationTypes';

const actions = {
  setMobile({ commit }) {
    commit(mutationTypes.SWITCH_TO_MOBILE);
  },

  setDesktop({ commit }) {
    commit(mutationTypes.SWITCH_TO_DESKTOP);
  }
};

export default actions;
