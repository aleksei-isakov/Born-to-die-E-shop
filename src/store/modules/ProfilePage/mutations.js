import mutationTypes from './mutationTypes';

const mutations = {
  [mutationTypes.SWITCH_TO_MOBILE](state) {
    state.isDesktop = false;
  },

  [mutationTypes.SWITCH_TO_DESKTOP](state) {
    state.isDesktop = true;
  }
};

export default mutations;
