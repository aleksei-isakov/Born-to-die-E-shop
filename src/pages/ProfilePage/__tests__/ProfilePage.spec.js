import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import ProfilePageMenu from '@/components/ProfilePageMenu/ProfilePageMenu.vue';
import ProfilePage from '@/pages/ProfilePage/ProfilePage';

let state;
let wrapper;
let store;
let actions;
let getters;
const localVue = createLocalVue();

localVue.use(Vuex);

beforeEach(() => {
  getters = {
    isDesktop: () => true
  };

  store = new Vuex.Store({
    modules: {
      ProfilePageModule: {
        actions,
        getters,
        state,
        namespaced: true
      }
    }
  });

  wrapper = shallowMount(ProfilePage, {
    stubs: {
      ProfilePageMenu: ProfilePageMenu
    },

    store,
    localVue
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('ProfilePage', () => {
  it('should render profile page for the user', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain desktop and mobile versions for the user', () => {
    expect(wrapper.find('profile-page-menu').exists());
  });
});
