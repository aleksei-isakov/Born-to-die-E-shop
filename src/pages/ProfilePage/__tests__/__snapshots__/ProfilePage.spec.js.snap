// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage should render profile page for the user 1`] = `
<div class="profile-page__wrapper">
  <h1 class="profile-page__header">Profile</h1>
  <div class="profile-page__sections">
    <div>
      <profile-page-menu-desktop-stub></profile-page-menu-desktop-stub>
    </div>
    <div class="profile-page__content"></div>
  </div>
</div>
`;
